yt_dlp/extractor/vimeo.py

```diff
@@ -1577,44 +1637,33 @@ def _fetch_page(self, album_id, authorization, hashed_pass, page):
-            video_id = self._search_regex(r'/videos/(\d+)', uri, 'video_id', default=None) if uri else None
+            video_id = self._search_regex(r'/videos/(\d+)', uri, 'id', default=None) if uri else None
+            if is_embed:
+                if not video_id:
+                    self.report_warning(f'Skipping due to missing video ID: {link}')
+                    continue
+                link = f'https://player.vimeo.com/video/{video_id}'
+                if referer:
+                    link = self._smuggle_referrer(link, referer)
             yield self.url_result(link, VimeoIE.ie_key(), video_id)
 
     def _real_extract(self, url):
-        album_id = self._match_id(url)
-        viewer = self._fetch_viewer_info(album_id, fatal=False)
-        if not viewer:
-            webpage = self._download_webpage(url, album_id)
-            viewer = self._parse_json(self._search_regex(
-                r'bootstrap_data\s*=\s*({.+?})</script>',
-                webpage, 'bootstrap data'), album_id)['viewer']
-        jwt = viewer['jwt']
-        album = self._download_json(
-            'https://api.vimeo.com/albums/' + album_id,
-            album_id, headers={'Authorization': 'jwt ' + jwt, 'Accept': 'application/json'},
-            query={'fields': 'description,name,privacy'})
-        hashed_pass = None
-        if try_get(album, lambda x: x['privacy']['view']) == 'password':
-            password = self.get_param('videopassword')
-            if not password:
-                raise ExtractorError(
-                    'This album is protected by a password, use the --video-password option',
-                    expected=True)
-            try:
-                hashed_pass = self._download_json(
-                    f'https://vimeo.com/showcase/{album_id}/auth',
-                    album_id, 'Verifying the password', data=urlencode_postdata({
-                        'password': password,
-                        'token': viewer['xsrft'],
-                    }), headers={
-                        'X-Requested-With': 'XMLHttpRequest',
-                    })['hashed_pass']
-            except ExtractorError as e:
-                if isinstance(e.cause, HTTPError) and e.cause.status == 401:
-                    raise ExtractorError('Wrong password', expected=True)
-                raise
+        url, _, http_headers = self._unsmuggle_headers(url)
+        album_id, is_embed = self._match_valid_url(url).group('id', 'is_embed')
+        referer = http_headers.get('Referer')
+
+        if not re.fullmatch(r'[0-9]+', album_id):
+            auth_info = self._download_json(
+                f'https://vimeo.com/showcase/{album_id}/auth', album_id, 'Downloading album info',
+                headers={'X-Requested-With': 'XMLHttpRequest', 'Referer': referer},
+                expected_status=(401, 403))
+            album_id = traverse_obj(auth_info, (
+                'metadata', 'id', {int}, {str_or_none}, {require('album ID')}))
```

Comment:
maybe we should always make this request so we can parse the `error_code` and raise w/ a proper msg when referer is required but missing